frontend/src/Editor/LeftSidebar/SidebarDatasources.jsx

```diff
@@ -69,42 +91,76 @@ export const LeftSidebarDataSources = ({
     return DataSourceTypes.find((source) => source.kind === dataSource.kind);
   };
 
-  const renderDataSource = (dataSource, idx) => {
+  const RenderDataSource = ({ dataSource, idx, convertToGlobal, showDeleteIcon = true, enableEdit = true }) => {
+    const [isConversionVisible, setConversionVisible] = React.useState(false);
     const sourceMeta = getSourceMetaData(dataSource);
 
     const icon = getSvgIcon(sourceMeta?.kind?.toLowerCase(), 24, 24, dataSource?.plugin?.iconFile?.data);
 
+    const convertToGlobalDataSource = (dataSource) => {
+      setConversionVisible(false);
+      changeScope(dataSource);
+    };
+
+    const popover = (
+      <PopoverBS id="popover-contained">
+        <PopoverBS.Body className={`${darkMode && 'theme-dark'}`}>
+          <div className={`row cursor-pointer`}>
+            <div className="col text-truncate cursor-pointer" onClick={() => convertToGlobalDataSource(dataSource)}>
+              Change scope
+            </div>
+          </div>
+        </PopoverBS.Body>
+      </PopoverBS>
+    );
+
     return (
       <div className="row mb-3 ds-list-item" key={idx}>
         <div
           role="button"
-          onClick={() => {
-            setSelectedDataSource(dataSource);
-            toggleDataSourceManagerModal(true);
-          }}
+          onClick={
+            enableEdit
+              ? () => {
+                  setSelectedDataSource(dataSource);
+                  toggleDataSourceManagerModal(true);
+                }
+              : null
+          }
           className="col d-flex align-items-center"
         >
           {icon}
           <span className="font-400" style={{ paddingLeft: 5 }}>
             {dataSource.name}
           </span>
         </div>
-        <div className="col-auto">
-          <button className="btn btn-sm p-1 ds-delete-btn" onClick={() => deleteDataSource(dataSource)}>
-            <div>
-              <TrashIcon width="14" height="14" />
-            </div>
-          </button>
-        </div>
+        {showDeleteIcon && (
+          <div className="col-auto">
+            <button className="btn btn-sm p-1 ds-delete-btn" onClick={() => deleteDataSource(dataSource)}>
+              <div>
+                <TrashIcon width="14" height="14" />
+              </div>
+            </button>
+          </div>
+        )}
+        {convertToGlobal && admin && (
+          <div className="col-auto">
+            <OverlayTrigger show={isConversionVisible} rootClose trigger="click" placement="bottom" overlay={popover}>
+              <div onClick={() => setConversionVisible(!isConversionVisible)}>
+                <VerticalIcon />
+              </div>
+            </OverlayTrigger>
+          </div>
+        )}
       </div>
     );
   };
 
   const popoverContent = (
     <LeftSidebarDataSources.Container
       darkMode={darkMode}
-      renderDataSource={renderDataSource}
+      RenderDataSource={RenderDataSource}
```

Comment:
Can we change it to camelCase and call it with a function? Looks bit odd with `capitalisation`